gui/include/euRun.hh

```diff
@@ -31,126 +31,142 @@ inline std::string to_bytes(const std::string & val) {
-      m_prevtrigs = 0;
-      m_prevtime = 0.0;
-      m_runstarttime = 0.0;
-      StartRun(cont ? "Continued" : txtRunmsg->displayText().toStdString());
-      EmitStatus("RUN", to_string(m_runnumber));
-      emit StatusChanged("EVENT", "0");
-      emit StatusChanged("TRIG", "0");
-      emit StatusChanged("PARTICLES", "0");
-      emit StatusChanged("RATE", "");
-      emit StatusChanged("MEANRATE", "");
-      SetState(ST_RUNNING);
-    }
-    void on_btnStop_clicked() {
-      StopRun();
-      EmitStatus("RUN", "(" + to_string(m_runnumber) + ")");
-      SetState(ST_READY);
-    }
-    void on_btnLog_clicked() {
-      std::string msg = txtLogmsg->displayText().toStdString();
-      EUDAQ_USER(msg);
     }
-    void timer() {
-      if (!m_stopping) {
-        if (m_runsizelimit >= 1024 && m_filebytes >= m_runsizelimit) {
-          EUDAQ_INFO("File limit reached: " + to_string(m_filebytes) + " > " + to_string(m_runsizelimit));
-          eudaq::mSleep(1000);
-          StopRun(false);
-          eudaq::mSleep(8000);
-          while (m_producerbusy) {
-            eudaq::mSleep(50);
-          }
-          on_btnStart_clicked(true);
-        } else if (dostatus) {
-          GetStatus();
-        }
-      }
+    if (m_startrunwhenready && !m_producer_pALPIDEfs_not_ok && !m_producer_pALPIDEss_not_ok) {
+      m_startrunwhenready = false;
+      on_btnStart_clicked(false);
     }
-    void ChangeStatus(const QString & name, const QString & value) {
-      status_t::iterator i = m_status.find(name.toStdString());
-      if (i != m_status.end()) {
-        i->second->setText(value);
-      }
+  }
+  void ChangeStatus(const QString & name, const QString & value) {
+    status_t::iterator i = m_status.find(name.toStdString());
+    if (i != m_status.end()) {
+      i->second->setText(value);
     }
-	void btnLogSetStatusSlot(bool status){
+  }
+  void btnLogSetStatusSlot(bool status){
 
-		 btnLog->setEnabled(status);
-	}
+    btnLog->setEnabled(status);
+  }
 	
 signals:
-    void StatusChanged(const QString &, const QString &);
-	void btnLogSetStatus(bool status);
-	void SetState(int status);
-  private:
-    RunControlModel m_run;
-    RunConnectionDelegate m_delegate;
-    QTimer m_statustimer;
-    typedef std::map<std::string, QLabel *> status_t;
-    status_t m_status;
-    int m_prevtrigs;
-    double m_prevtime, m_runstarttime;
-    int64_t m_filebytes;
-    bool dostatus;
+  void StatusChanged(const QString &, const QString &);
+  void btnLogSetStatus(bool status);
+  void SetState(int status);
+private:
+  RunControlModel m_run;
+  RunConnectionDelegate m_delegate;
+  QTimer m_statustimer;
+  typedef std::map<std::string, QLabel *> status_t;
+  status_t m_status;
+  int m_prevtrigs;
+  double m_prevtime, m_runstarttime;
+  long long m_filebytes;
+  long long m_events;
+  bool dostatus;
+  bool m_producer_pALPIDEfs_not_ok;
+  bool m_producer_pALPIDEss_not_ok;
+  bool m_startrunwhenready;
```

Comment:
Please refrain from changing indentation since this makes it nearly impossible to detect changes. We are aware of the fact that some code is really badly layouted and we are planning to run a beautification tool before releasing 1.5.